language_formatters_pre_commit_hooks/pretty_format_yaml.py

```diff
@@ -47,14 +47,17 @@ def pretty_format_yaml(argv=None):
             string_content = ''.join(f.readlines())
 
         try:
-            content = yaml.load(string_content)
+            content = list(yaml.load_all(string_content))
+            pretty_content = StringIO()
 
-            if not isinstance(content, (list, dict)):
-                # skip files containing primitive types (unstructured text)
-                continue
+            if len(content) == 1:
+                if not isinstance(content[0], (list, dict)):
+                    # skip files containing primitive types (unstructured text)
+                    continue
 
-            pretty_content = StringIO()
-            yaml.dump(content, pretty_content)
+                yaml.dump(content[0], pretty_content)
+            else:
+                yaml.dump_all(content, pretty_content)
```

Comment:
The check that excludes from pretty-fication the primitive types needs to be done for all the documents (more context in [here](https://github.com/macisamuele/language-formatters-pre-commit-hooks/pull/1#discussion_r370216799)).

I do believe that the current implementation has the potential of creating a new occurrence of #1 .

In order to work around this issue I think that we might pretty-format each individual document and then concat them. Something like the following
```python
pretty_docs = []
for ydoc in yaml.load_all(...):
    if isinstance(ydoc, (list, dict)):
        pretty_content = StringIO()
        yaml.dump(ydoc, pretty_content)
        pretty_docs.append(pretty_content.getvalue())
    else:
        # skip files containing primitive types (unstructured text)	
        pretty_docs.append(str(ydoc))

output = '\n---'.join(pretty_docs)
```
(disclaimer: I have not actually tried the code, so it could be very buggy)

Reply:
NOTE: You might try to visualise the mentioned issue by running the following code
```python
from io import StringIO
from ruamel.yaml import YAML
yaml_content  = "---\n{}\n---\na: 1".format('a ' * 100)

fixed_content = StringIO()
YAML().dump_all(documents=list(YAML().load_all(yaml_content)), stream=fixed_content)

print("Initial content:")
print(yaml_content)
print("Fixed content:")
print(fixed_content.getvalue())

assert YAML().load_all(yaml_content) == YAML().load_all(fixed_content.getvalue())
```
Output
```
Initial content:
---
a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a
---
a: 1
Fixed content:
a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a
a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a a
a a a a a a a a a a a a a a a a a a
---
a: 1

Traceback (most recent call last):
  File "a.py", line 13, in <module>
    assert YAML().load_all(yaml_content) == YAML().load_all(fixed_content.getvalue())
AssertionError
```
You can see how the first document is different (and as it is a string it is actually a different string

FYI I've noticed that this phenomena does actually happen also if the string is used as value of a mapping (#4)

Reply:
I thought about this, but I'm not sure it's legit to have a real YAML file that contains non-YAML (primitive) documents.  Can you point me to a real-world example of this?

Reply:
I don't really have an example here, personally I use YAML mostly for swagger specs or writing configurations (so always mappings and single docs).

The specifications do actually allow to have primitive types at the root level and so the following file should be considered valid
```yaml
---
null
---
this is a string
---
1
---
true
---
1.0
---
- first item
- second item
---
key1: value 1
key2: value 2
```

Considering this, I think that we should handle such case as well (especially considering that the effort to do so is not huge)

Reply:
OK, fair enough. Shouldn't be a problem to add it and maybe include a test or two as well.